web_chat.lua

```diff
@@ -103,8 +104,38 @@ local ltInfo          = 2
 local ltWarning       = 3
 local ltError         = 4
 
+-- Adds Webchat callback, plugins can return true to 
+-- prevent message from appearing / being processed
+-- by further callbacks
+-- OnWebChat(Username, Message)
+function AddWebChatCallback(PluginName, FunctionName) 
+	
+	for k, v in pairs(OnWebChatCallbacks) do
+		if v[1] == PluginName and v[2] == FunctionName then
+			return false
+		end
+	end
+	
+	table.insert(OnWebChatCallbacks, {PluginName, FunctionName})
+	return true
+end
 
-
+-- Removes webchat callback
+function RemoveWebChatCallback(PluginName, FunctionName) 
+	
+	local i=1
+	while i <= #OnWebChatCallbacks do
```

Comment:
Go the other way round - from `#OnWebChatCallbacks` down to 1. This will save some performance, as counting the number of array items is slow in Lua.